src/Cis194/Hw/LogAnalysis.hs

```diff
@@ -6,22 +6,44 @@ module Cis194.Hw.LogAnalysis where
 import  Cis194.Hw.Log
 
 parseMessage :: String -> LogMessage
-parseMessage s = Unknown s
+parseMessage s =
+  case words s of
+    "E":i:ws -> buildMessage (Error (read i)) ws
+    "I":ws -> buildMessage Info ws
+    "W":ws -> buildMessage Warning ws
+    _ -> Unknown s
+
+buildMessage :: MessageType -> [String] -> LogMessage
+buildMessage m (t:ws) = LogMessage m (read t) (unwords ws)
 
 parse :: String -> [LogMessage]
-parse _ = []
+parse = map (parseMessage) . lines
 
 insert :: LogMessage -> MessageTree -> MessageTree
+insert (Unknown _) t = t
```

Comment:
As written this pattern is just fine, but a neat thing you can use is record constructor syntax to match a constructor where you don't care about any of its fields. This pays off more for constructors with more fields:

```hs
insert Unknown{} t = t
```